src/yardstick/cli/result.py

```diff
@@ -176,41 +257,55 @@ def list_tools(cfg: config.Application, result_set: str):
     for result in results:
         tools.add(result.tool)
 
-    for tool in sorted(list(tools)):
+    for tool in sorted(tools):
         print(tool)
 
 
 @dataclass(eq=True, order=True)
 class ResultDescription:
     image: str
     tool: str
-    timestamp: str
     ID: str
+    timestamp: datetime.datetime | None = None
 
 
-def result_descriptions(result_set: Optional[str] = None) -> list[ResultDescription]:
+def result_descriptions(result_set: str | None = None) -> list[ResultDescription]:
     results = []
     if result_set:
         result_set_obj = store.result_set.load(result_set)
 
         for result_state in result_set_obj.state:
-            scan_config = store.scan_result.find_one(by_description=result_state.config.path)
+            if not result_state.config:
+                continue
+
+            scan_config = store.scan_result.find_one(
+                by_description=result_state.config.path,
+            )
+
+            if not scan_config:
```

Comment:
This seems like it didn't raise before the linter update. Is that correct? I'm concerned with a change this large introducing a new exception. Why is it correct to continue on line 279 but raise here?

Reply:
mypi found this one. Technically `result_state.config` is optional, and trying to search when it's missing will cause an exception. The continue is a guard clause for this case. 

On the second case we could not find a result given then input, so all paths moving forward are not valid, which is why we raise.

The second behavior is correct, however, I don't have a solid reason for the continue statement for the first part... I think moving it to a `raise` instead of continue makes sense. I'll update it